### PR TITLE
NAS-119380 / 23.10 / Do not treat `legacy` as mountpoint

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
time 0:10

**Comment by yocalebo** [comment]
Should we really crash here? This will break `pool.dataset.unlock` when `toggle_attachments` is set.

**Comment by themylogin** [comment]
@yocalebo why will it break? We don't have legacy mountpoints anywhere besides boot-pool which is not affected

**Review by Ornias1993 — COMMENTED** [review]
I'm not 100% sure if this would trigger this codepath, but a LOT of App related paths are, in fact, legacy....

File: src/middlewared/middlewared/plugins/pool_/dataset_processes.py [hunk 1]
```diff
@@ -41,9 +41,12 @@ async def processes(self, oid):
         dataset = await self.middleware.call('pool.dataset.get_instance_quick', oid, {'encryption': True})
         if dataset['locked']:
             return []
-        path = attachments_path(dataset)
-        zvol_path = f'/dev/zvol/{dataset["name"]}'
-        return await self.middleware.call('pool.dataset.processes_using_paths', [path, zvol_path])
+
+        paths = [f'/dev/zvol/{dataset["name"]}']
```
Comment (anodos325) [inline src/middlewared/middlewared/plugins/pool_/dataset_processes.py:45]: Hmm.... is this a place where we missed conversion to using `zvol_name_to_path()` ? 

https://github.com/truenas/middleware/blob/master/src/middlewared/middlewared/plugins/zfs_/utils.py#L19

Comment (themylogin) [reply src/middlewared/middlewared/plugins/pool_/dataset_processes.py:45]: Great catch! We actually had a few places like this

**Review by anodos325 — APPROVED** [review]

**Pull request merged**